adam/learner/contrastive_learner.py

```diff
@@ -244,69 +265,83 @@ def log_hypotheses(self, log_output_path: Path) -> None:
     def _update_counts(
         self,
         concept: Concept,
-        pattern_to_graph_match: PerceptionGraphPatternMatch,
+        pattern_to_graph_matches: Mapping[
```

Comment:
Since we only use the values here, I prefer to make this `Sequence[Optional[PerceptionGraphPatternMatch]]` Or better, `Sequence[PerceptionGraphPatternMatch]` and just filter out the `None` s at the call site.

Reply:
Ok, without altering the `_match_concept_pattern_to_multiple_graphs` function, I've extracted the values into a new variable and parsed the `None`s out beforehand.